### PR TITLE
chore(runtime): Update weights

### PR DESCRIPTION
**Pull request by breathx**
### Comparison table for Vara runtime for Instruction

| name                    | master  | dn-update-weights | diff    |
|-------------------------|---------|-------------------|---------|
| i64const                | 126 ps  | 155 ps            | +18.71% |
| i64load                 | 11.4 ns | 9.5 ns            | -19.60% |
| i32load                 | 11.1 ns | 9.2 ns            | -20.91% |
| i64store                | 19.4 ns | 23.7 ns           | +18.03% |
| i32store                | 15.4 ns | 14.2 ns           | -8.52%  |
| select                  | 7.4 ns  | 7.3 ns            | -1.29%  |
| if                      | 6.3 ns  | 6.3 ns            | +0.35%  |
| br                      | 3.1 ns  | 3.2 ns            | +1.29%  |
| br_if                   | 5.7 ns  | 5.9 ns            | +3.52%  |
| br_table                | 10.0 ns | 9.3 ns            | -7.11%  |
| br_table_per_entry      | 344 ps  | 361 ps            | +4.71%  |
| call                    | 4.6 ns  | 4.5 ns            | -2.25%  |
| call_indirect           | 19.5 ns | 20.2 ns           | +3.37%  |
| call_indirect_per_param | 2.3 ns  | 2.2 ns            | -3.97%  |
| call_per_local          | 0 ps    | 0 ps              | N/A     |
| local_get               | 777 ps  | 1.1 ns            | +29.23% |
| local_set               | 1.9 ns  | 2.6 ns            | +25.23% |
| local_tee               | 1.9 ns  | 2.9 ns            | +32.72% |
| global_get              | 1.6 ns  | 1.9 ns            | +12.69% |
| global_set              | 2.6 ns  | 2.8 ns            | +7.16%  |
| memory_current          | 14.5 ns | 14.4 ns           | -0.98%  |
| i64clz                  | 6.7 ns  | 6.8 ns            | +1.32%  |
| i32clz                  | 6.3 ns  | 6.3 ns            | +0.64%  |
| i64ctz                  | 6.2 ns  | 6.2 ns            | +0.56%  |
| i32ctz                  | 5.3 ns  | 5.3 ns            | +0.15%  |
| i64popcnt               | 1.1 ns  | 1.2 ns            | +8.16%  |
| i32popcnt               | 724 ps  | 746 ps            | +2.95%  |
| i64eqz                  | 3.7 ns  | 3.7 ns            | -0.38%  |
| i32eqz                  | 2.3 ns  | 2.4 ns            | +3.24%  |
| i32extend8s             | 780 ps  | 976 ps            | +20.08% |
| i32extend16s            | 851 ps  | 885 ps            | +3.84%  |
| i64extend8s             | 981 ps  | 993 ps            | +1.21%  |
| i64extend16s            | 972 ps  | 1.0 ns            | +6.09%  |
| i64extend32s            | 924 ps  | 936 ps            | +1.28%  |
| i64extendsi32           | 673 ps  | 866 ps            | +22.29% |
| i64extendui32           | 324 ps  | 556 ps            | +41.73% |
| i32wrapi64              | 213 ps  | 346 ps            | +38.44% |
| i64eq                   | 3.6 ns  | 3.6 ns            | +0.00%  |
| i32eq                   | 2.0 ns  | 2.3 ns            | +11.49% |
| i64ne                   | 3.5 ns  | 3.5 ns            | +1.69%  |
| i32ne                   | 2.0 ns  | 2.3 ns            | +13.02% |
| i64lts                  | 3.4 ns  | 3.6 ns            | +5.75%  |
| i32lts                  | 2.0 ns  | 2.4 ns            | +19.35% |
| i64ltu                  | 3.4 ns  | 3.9 ns            | +11.32% |
| i32ltu                  | 2.1 ns  | 2.5 ns            | +14.01% |
| i64gts                  | 3.5 ns  | 4.0 ns            | +11.51% |
| i32gts                  | 2.0 ns  | 2.4 ns            | +16.09% |
| i64gtu                  | 3.6 ns  | 3.8 ns            | +5.31%  |
| i32gtu                  | 2.1 ns  | 2.5 ns            | +14.89% |
| i64les                  | 3.6 ns  | 3.6 ns            | +0.69%  |
| i32les                  | 2.3 ns  | 2.3 ns            | -0.75%  |
| i64leu                  | 3.7 ns  | 3.5 ns            | -4.55%  |
| i32leu                  | 2.2 ns  | 2.3 ns            | +4.94%  |
| i64ges                  | 3.7 ns  | 3.5 ns            | -5.05%  |
| i32ges                  | 2.3 ns  | 2.2 ns            | -5.25%  |
| i64geu                  | 3.7 ns  | 3.6 ns            | -1.94%  |
| i32geu                  | 2.1 ns  | 2.2 ns            | +3.86%  |
| i64add                  | 2.5 ns  | 2.6 ns            | +1.82%  |
| i32add                  | 1.1 ns  | 1.2 ns            | +11.20% |
| i64sub                  | 2.5 ns  | 2.7 ns            | +5.15%  |
| i32sub                  | 1.2 ns  | 1.1 ns            | -10.09% |
| i64mul                  | 3.5 ns  | 3.5 ns            | +1.75%  |
| i32mul                  | 2.4 ns  | 2.3 ns            | -2.00%  |
| i64divs                 | 4.1 ns  | 3.3 ns            | -23.58% |
| i32divs                 | 4.0 ns  | 3.7 ns            | -9.02%  |
| i64divu                 | 4.6 ns  | 4.4 ns            | -4.17%  |
| i32divu                 | 4.0 ns  | 4.2 ns            | +4.30%  |
| i64rems                 | 16.4 ns | 15.5 ns           | -5.95%  |
| i32rems                 | 13.5 ns | 13.3 ns           | -1.30%  |
| i64remu                 | 4.5 ns  | 4.1 ns            | -9.08%  |
| i32remu                 | 4.1 ns  | 4.2 ns            | +1.79%  |
| i64and                  | 2.7 ns  | 2.7 ns            | -1.78%  |
| i32and                  | 1.3 ns  | 1.2 ns            | -14.62% |
| i64or                   | 2.6 ns  | 2.6 ns            | -1.90%  |
| i32or                   | 1.2 ns  | 1.2 ns            | -2.72%  |
| i64xor                  | 2.6 ns  | 2.7 ns            | +2.07%  |
| i32xor                  | 1.3 ns  | 1.3 ns            | -1.67%  |
| i64shl                  | 2.1 ns  | 2.1 ns            | -0.80%  |
| i32shl                  | 1.0 ns  | 1.1 ns            | +3.04%  |
| i64shrs                 | 2.2 ns  | 2.2 ns            | +1.83%  |
| i32shrs                 | 1.1 ns  | 1.2 ns            | +9.97%  |
| i64shru                 | 2.2 ns  | 2.4 ns            | +8.95%  |
| i32shru                 | 1.1 ns  | 1.1 ns            | +2.02%  |
| i64rotl                 | 2.1 ns  | 2.2 ns            | +5.24%  |
| i32rotl                 | 1.0 ns  | 1.3 ns            | +20.27% |
| i64rotr                 | 2.1 ns  | 2.3 ns            | +7.72%  |
| i32rotr                 | 1.1 ns  | 1.2 ns            | +15.51% |

### Comparison table for Vara runtime for HostFn

| name                                    | master   | dn-update-weights | diff     |
|-----------------------------------------|----------|-------------------|----------|
| alloc                                   | 7.0 µs   | 7.2 µs            | +2.57%   |
| alloc_per_page                          | 314.8 ns | 355.7 ns          | +11.49%  |
| free                                    | 787.9 ns | 757.9 ns          | -3.96%   |
| free_range                              | 953.4 ns | 901.0 ns          | -5.82%   |
| free_range_per_page                     | 66.3 ns  | 61.6 ns           | -7.65%   |
| gr_reserve_gas                          | 2.3 µs   | 2.4 µs            | +1.47%   |
| gr_unreserve_gas                        | 2.3 µs   | 2.2 µs            | -4.50%   |
| gr_system_reserve_gas                   | 1.2 µs   | 1.2 µs            | -4.27%   |
| gr_gas_available                        | 1.2 µs   | 1.1 µs            | -12.64%  |
| gr_message_id                           | 1.1 µs   | 1.1 µs            | +1.55%   |
| gr_program_id                           | 1.1 µs   | 1.1 µs            | -3.91%   |
| gr_source                               | 1.2 µs   | 1.0 µs            | -10.18%  |
| gr_value                                | 1.1 µs   | 1.1 µs            | -4.32%   |
| gr_value_available                      | 1.1 µs   | 1.1 µs            | -4.87%   |
| gr_size                                 | 1.1 µs   | 1.1 µs            | -1.81%   |
| gr_read                                 | 1.8 µs   | 1.8 µs            | +0.06%   |
| gr_read_per_byte                        | 164 ps   | 165 ps            | +0.61%   |
| gr_env_vars                             | 1.9 µs   | 1.3 µs            | -45.31%  |
| gr_block_height                         | 1.1 µs   | 1.1 µs            | -2.54%   |
| gr_block_timestamp                      | 1.2 µs   | 1.1 µs            | -8.02%   |
| gr_random                               | 2.2 µs   | 2.1 µs            | -3.30%   |
| gr_reply_deposit                        | 6.6 µs   | 6.5 µs            | -1.20%   |
| gr_send                                 | 3.2 µs   | 3.2 µs            | -2.37%   |
| gr_send_per_byte                        | 259 ps   | 271 ps            | +4.43%   |
| gr_send_wgas                            | 3.3 µs   | 3.2 µs            | -2.68%   |
| gr_send_wgas_per_byte                   | 258 ps   | 269 ps            | +4.09%   |
| gr_send_init                            | 1.2 µs   | 1.2 µs            | -1.38%   |
| gr_send_push                            | 2.2 µs   | 2.1 µs            | -6.05%   |
| gr_send_push_per_byte                   | 379 ps   | 382 ps            | +0.79%   |
| gr_send_commit                          | 2.8 µs   | 2.7 µs            | -3.85%   |
| gr_send_commit_wgas                     | 2.9 µs   | 2.7 µs            | -4.95%   |
| gr_reservation_send                     | 3.5 µs   | 3.4 µs            | -2.99%   |
| gr_reservation_send_per_byte            | 267 ps   | 267 ps            | +0.00%   |
| gr_reservation_send_commit              | 3.1 µs   | 2.9 µs            | -5.11%   |
| gr_reply_commit                         | 18.6 µs  | 16.0 µs           | -16.35%  |
| gr_reply_commit_wgas                    | 18.0 µs  | 21.4 µs           | +15.94%  |
| gr_reservation_reply                    | 9.3 µs   | 9.3 µs            | -0.06%   |
| gr_reservation_reply_per_byte           | 431.5 ns | 432.2 ns          | +0.17%   |
| gr_reservation_reply_commit             | 13.3 µs  | 7.2 µs            | -84.70%  |
| gr_reply_push                           | 2.0 µs   | 1.8 µs            | -8.43%   |
| gr_reply                                | 20.7 µs  | 18.4 µs           | -12.25%  |
| gr_reply_per_byte                       | 417 ps   | 417 ps            | +0.00%   |
| gr_reply_wgas                           | 18.6 µs  | 17.6 µs           | -6.24%   |
| gr_reply_wgas_per_byte                  | 422 ps   | 423 ps            | +0.24%   |
| gr_reply_push_per_byte                  | 660 ps   | 675 ps            | +2.22%   |
| gr_reply_to                             | 1.2 µs   | 1.1 µs            | -6.56%   |
| gr_signal_code                          | 1.1 µs   | 1.1 µs            | -4.39%   |
| gr_signal_from                          | 1.1 µs   | 1.1 µs            | -5.11%   |
| gr_reply_input                          | 13.7 µs  | 33.6 µs           | +59.34%  |
| gr_reply_input_wgas                     | 54.8 µs  | 0 ps              | N/A      |
| gr_reply_push_input                     | 1.4 µs   | 1.4 µs            | -2.67%   |
| gr_reply_push_input_per_byte            | 166 ps   | 163 ps            | -1.84%   |
| gr_send_input                           | 3.4 µs   | 3.4 µs            | -0.48%   |
| gr_send_input_wgas                      | 3.5 µs   | 3.6 µs            | +1.75%   |
| gr_send_push_input                      | 1.7 µs   | 1.6 µs            | -1.77%   |
| gr_send_push_input_per_byte             | 167 ps   | 170 ps            | +1.76%   |
| gr_debug                                | 1.5 µs   | 1.5 µs            | +0.03%   |
| gr_debug_per_byte                       | 316 ps   | 321 ps            | +1.56%   |
| gr_reply_code                           | 1.1 µs   | 1.1 µs            | -6.14%   |
| gr_exit                                 | 196.2 µs | 197.4 µs          | +0.62%   |
| gr_leave                                | 188.5 µs | 191.2 µs          | +1.40%   |
| gr_wait                                 | 124.7 µs | 127.7 µs          | +2.38%   |
| gr_wait_for                             | 191.3 µs | 191.9 µs          | +0.32%   |
| gr_wait_up_to                           | 186.1 µs | 188.4 µs          | +1.22%   |
| gr_wake                                 | 2.2 µs   | 2.0 µs            | -5.26%   |
| gr_create_program                       | 4.3 µs   | 4.3 µs            | -0.94%   |
| gr_create_program_payload_per_byte      | 261 ps   | 86 ps             | -203.49% |
| gr_create_program_salt_per_byte         | 2.1 ns   | 1.9 ns            | -5.92%   |
| gr_create_program_wgas                  | 4.4 µs   | 4.3 µs            | -1.84%   |
| gr_create_program_wgas_payload_per_byte | 254 ps   | 87 ps             | -191.95% |
| gr_create_program_wgas_salt_per_byte    | 2.1 ns   | 1.9 ns            | -6.96%   |

### Comparison table for Vara runtime for Memory

| name                                  | master   | dn-update-weights | diff    |
|---------------------------------------|----------|-------------------|---------|
| lazy_pages_signal_read                | 28.6 µs  | 29.1 µs           | +1.66%  |
| lazy_pages_signal_write               | 34.9 µs  | 36.0 µs           | +3.12%  |
| lazy_pages_signal_write_after_read    | 9.8 µs   | 10.6 µs           | +7.47%  |
| lazy_pages_host_func_read             | 29.8 µs  | 30.9 µs           | +3.63%  |
| lazy_pages_host_func_write            | 34.2 µs  | 37.4 µs           | +8.78%  |
| lazy_pages_host_func_write_after_read | 9.1 µs   | 9.2 µs            | +0.52%  |
| load_page_data                        | 10.3 µs  | 10.3 µs           | -0.32%  |
| upload_page_data                      | 104.1 µs | 104.0 µs          | -0.11%  |
| static_page                           | 100 ps   | 100 ps            | +0.00%  |
| mem_grow                              | 933.0 ns | 1.3 µs            | +28.09% |
| parachain_read_heuristic              | 0 ps     | 0 ps              | N/A     |

### CODE REVIEW

**Comment by ukint-vs** [comment]
Interesting what happened with 'gr_reservation_reply_commit'. Maybe there some optimisation caused by the new wasmtime.

**Review by ukint-vs — APPROVED** [review]

File: pallets/gear/src/schedule.rs [hunk 1]
```diff
@@ -761,7 +761,7 @@ impl Default for Limits {
 impl<T: Config> Default for InstructionWeights<T> {
     fn default() -> Self {
         Self {
-            version: 11,
+            version: 1050,
```
Comment (ukint-vs) [inline pallets/gear/src/schedule.rs:764]: I think we don't want to have to reinstrument programs after every release. It's pointless if nothing related to weight and performance is introduced.

**Review by ukint-vs — APPROVED** [review]

File: pallets/gear/src/schedule.rs (hunk 1 above)
Comment (breathx) [reply pallets/gear/src/schedule.rs:764]: Yep, you're right, so it's not supposed to bump every single time, but I want version to reflect release number when it was updated last time. How do you think?

Comment (ukint-vs) [reply pallets/gear/src/schedule.rs]: Fine with this.

**Review by gshep — APPROVED** [review]

**Pull request merged**